src/Curiosity/Data.hs

```diff
@@ -71,6 +75,17 @@ instantiateStmDb Db { _dbNextUserId = Identity seedNextUserId, _dbUserProfiles =
 instantiateEmptyStmDb :: forall runtime m . MonadIO m => m (StmDb runtime)
 instantiateEmptyStmDb = instantiateStmDb emptyHask
 
+-- | Reset all values of the `Db` product type from `STM.STM` to the empty
+-- state.
+resetStmDb
+  :: forall runtime m . MonadIO m => StmDb runtime -> m ()
+resetStmDb stmDb = liftIO . STM.atomically $ do
+  STM.writeTVar (_dbNextUserId stmDb) seedNextUserId
+  STM.writeTVar (_dbUserProfiles stmDb) seedProfiles
+  STM.writeTVar (_dbTodos stmDb) seedTodos
```

Comment:
Can the `seed` values be provided as an argument to the function as a `HaskDb` value? 
We can also have [`instance Default HaskDb`](https://hackage.haskell.org/package/data-default-class-0.1.2.0/docs/Data-Default-Class.html#t:Default) to make the usage easier (via `def`).

Reply:
The `where` clause below is using `emptyHask`. The problem is to destructure the `HaskDb` value to unwrap the values (wrapped in `Identity`). I'm not sure how to get a hold of those values without doing that. (B.t.w. I think I simply followed the way you did it somewhere else.)

It seems using the `emptyHask` value is the right way to go; it is more explicit than some "unknown" default. Or maybe I didn't really understand your point ?